browser/todo-list.md

```diff
@@ -3,291 +3,248 @@
 
-   ```bash
-    $ pwd 
-    /Users/meepers/repos/blockstack-todos
-   ```
-   
-   This path will be different for you, but double-check the last part to ensure that you're in the directory into which you cloned and in which you ran `npm install`.
+    // ...
 
-2. Start the application in your local environment.
+    const userSession = new UserSession({ appConfig })
 
-    ```bash
-    $ npm run start
     ```
-    
-    You should see output similar to the following:
 
-    ```bash
-      Compiled successfully!
+    This module handles user session operations and is initiated using the [`appConfig`](https://github.com/blockstack/blockstack-todos/blob/master/src/assets/constants.js#L3) object, which contains an array of [scopes](/develop/overview_auth.html#scopes) that indicate just what permissions to grant during authentication:
 
-      You can now view bs-todo in the browser.
+    ```js
+    export const appConfig = new AppConfig(['store_write', 'publish_data'])
+    ```
 
-        http://127.0.0.1:3000/
+    The `appDetails` and `userSession` objects are joined by the callback function [`finished`](https://github.com/blockstack/blockstack-todos/blob/master/src/components/App.jsx#L31) in configuring Blockstack Connect for authentication with the `authOptions` object:
 
-      Note that the development build is not optimized.
-      To create a production build, use npm run build.
-      ```
+    ```js
+    finished: ({ userSession }) => {
+      this.setState({ userData: userSession.loadUserData() });
+    }
 
-2. Open your local browser to the `http://localhost:3000` URL.
-  
-    You should see a simple application:
+    ```
 
-    ![](images/todo-sign-in.png)
+    This function simply saves data about the user into the app's state upon authentication.
 
- 3. Choose **Sign In with Blockstack**.
+    Further down in the component we see in [`componentDidMount`](https://github.com/blockstack/blockstack-todos/blob/master/src/components/App.jsx#L46) that it checks upon mount to either process completion of authentication with `userSession.handlePendingSignIn()` or otherwise load session data into app state as above with `userSession.isUserSignedIn()`:
 
-    If you have already signed into Blockstack the application prompts you to select the ID to use. If you aren’t signed in, Blockstack prompts you to:
+    ```js
+    componentDidMount() {
+      if (userSession.isSignInPending()) {
+        userSession.handlePendingSignIn().then((userData) => {
+          window.history.replaceState({}, document.title, "/")
+          this.setState({ userData: userData})
+        });
+      } else if (userSession.isUserSignedIn()) {
+        this.setState({ userData: userSession.loadUserData() });
+      }
+    }
+    ```
 
-    ![](images/login-choice.png)
 
-    If the login to the application is successful, the user is presented with the application:
+2. Choose **Get started** to generate a *Secret Key*.
```

Comment:
I think I was already asked to do this earlier, but I was not. I completed the flow while I was in the app already.. 

Maybe say something like "going back into the app, ..." in here. And explicitly say before that a user doesn't need to follow the steps on the screen *just yet*?